src/Compiler/SyntaxTree/LexFilter.fs

```diff
@@ -2019,7 +2118,13 @@ type LexFilterImpl (
             returnToken tokenLexbufState token
 
         //  module ... ~~~> CtxtModuleHead
+        //  Check for inappropriate nesting within type definitions
         | MODULE, _ :: _ ->
+            // Check if this module definition is inappropriately nested in a type
+            match "MODULE" with
+            | InvalidDeclsInTypeDefn errorMsg -> error tokenTup errorMsg
+            | _ -> ()
```

Comment:
Matching a literal looks suspicious. I think it would be easier to understand the intention if a normal function call was used here.